WikiWatershed/model-my-watershed
# Upgrade Windshaft

## Overview

Upgrades the Windshaft tiler.

Connects #3091 
Connects #3082 

### Demo

![image](https://user-images.githubusercontent.com/1430060/60213612-dd972380-9831-11e9-967c-b8d9d68be57f.png)

![2019-06-26 16 59 54](https://user-images.githubusercontent.com/1430060/60214770-251eaf00-9834-11e9-9766-91d3596ceace.gif)

### Notes
<details>
<summary>

~Currently getting the following error in the "Install Windshaft JavaScript Dependencies" step:~ fixed in 8c5d49a8.

</summary>

```json
{
    "changed": true,
    "cmd": [
        "npm",
        "install",
        "--unsafe-perm"
    ],
    "delta": "0:01:04.762965",
    "end": "2019-06-18 15:55:30.183979",
    "msg": "non-zero return code",
    "rc": 254,
    "start": "2019-06-18 15:54:25.421014",
    "stderr": "npm WARN deprecated queue-async@1.1.0: renamed to d3-queue\nnpm WARN deprecated sphericalmercator@1.0.5: This module is now under the @mapbox namespace: install @mapbox/sphericalmercator instead\nnpm WARN deprecated husl@6.0.6: Project renamed to HSLuv\nnpm WARN mmw_tiler@1.0.0 No description\nnpm WARN mmw_tiler@1.0.0 No repository field.\n\nnpm ERR! path /opt/tiler/node_modules/gdal/node_modules/rc/node_modules/minimist/package.json.2871486687\nnpm ERR! code ENOENT\nnpm ERR! errno -2\nnpm ERR! syscall open\nnpm ERR! enoent ENOENT: no such file or directory, open '/opt/tiler/node_modules/gdal/node_modules/rc/node_modules/minimist/package.json.2871486687'\nnpm ERR! enoent This is related to npm not being able to find a file.\nnpm ERR! enoent \n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /home/vagrant/.npm/_logs/2019-06-18T15_55_30_167Z-debug.log",
    "stderr_lines": [
        "npm WARN deprecated queue-async@1.1.0: renamed to d3-queue",
        "npm WARN deprecated sphericalmercator@1.0.5: This module is now under the @mapbox namespace: install @mapbox/sphericalmercator instead",
        "npm WARN deprecated husl@6.0.6: Project renamed to HSLuv",
        "npm WARN mmw_tiler@1.0.0 No description",
        "npm WARN mmw_tiler@1.0.0 No repository field.",
        "",
        "npm ERR! path /opt/tiler/node_modules/gdal/node_modules/rc/node_modules/minimist/package.json.2871486687",
        "npm ERR! code ENOENT",
        "npm ERR! errno -2",
        "npm ERR! syscall open",
        "npm ERR! enoent ENOENT: no such file or directory, open '/opt/tiler/node_modules/gdal/node_modules/rc/node_modules/minimist/package.json.2871486687'",
        "npm ERR! enoent This is related to npm not being able to find a file.",
        "npm ERR! enoent ",
        "",
        "npm ERR! A complete log of this run can be found in:",
        "npm ERR!     /home/vagrant/.npm/_logs/2019-06-18T15_55_30_167Z-debug.log"
    ],
    "stdout": "",
    "stdout_lines": []
}
```

</details>

Rollbar and S3 Caching will be tested once this is on staging.

## Testing Instructions

* Delete the `src/tiler/node_modules` directory in your host
* Check out this branch, destroy the `tiler` and reprovision it
* Ensure you have various stream / boundary datasets installed:

    ```bash
    $ vagrant ssh app -c 'cd /vagrant && ./scripts/aws/setupdb.sh -b -d -s -c -p -q -m'
    ```

* Go to [:8000/](http://localhost:8000/) and turn on each layer in the layer selector
  - [x] Ensure all tiler layers work
* Choose "Select Boundary" as the draw tool, and pick HUC-8 as the boundary type
* Hover around an area which is homogenous, i.e. has the same boundary value for entire tiles
* Search away from that area to another homogenous area, i.e. has a different, but same boundary value for entire tiles
  - [x] Ensure the hover value is of the correct boundary, and not a stale value
